python/cudf/cudf/core/column/column.py

```diff
@@ -1832,9 +1856,17 @@ def as_column(
                 dtype=arbitrary.dtype,
             )
         elif arb_dtype.kind in ("O", "U"):
-            data = as_column(
-                pa.Array.from_pandas(arbitrary), dtype=arbitrary.dtype
-            )
+            if isinstance(arbitrary[0], pd._libs.interval.Interval):
+                # changing from pd array to series,possible arrow bug
+                interval_series = pd.Series(arbitrary)
+                data = as_column(
+                    pa.Array.from_pandas(interval_series),
+                    dtype=arbitrary.dtype,
+                )
```

Comment:
What situation came up where we get a numpy array of `object` type with interval objects in it?

Reply:
This happens when we want to make a cudf Series of interval objects. The list of intervals is initially passed through `as_column` and on line 1998 it's changed to a np.array, that's then passed to `as_column` again. 
```
data = np.asarray(
                        arbitrary,
                        dtype=native_dtype
                        if native_dtype is None
                        else np.dtype(native_dtype),
                    )
```

Reply:
Thanks for the explanation 😄